src/cockpit/channels/stream.py

```diff
@@ -46,7 +46,7 @@ class UnixStreamChannel(ProtocolChannel):
     payload = 'stream'
     restrictions = (('unix', None),)
 
-    def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, object]) -> SocketTransport:
+    async def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, object]) -> SocketTransport:
         path: str = options['unix']
         connection = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
```

Comment:
It's probably worth replacing this with the non-blocking equivalent.

Reply:
In fact, after you do that, you might find that the unix channel and the tcp channel end up looking fairly similar...

Reply:
*nod*, I thought the same, I just wanted to wait for a first round of CI. I merged both into a single class now. This has the advantage that we can give better error messages for wrong options.